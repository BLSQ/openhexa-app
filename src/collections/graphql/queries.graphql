query CollectionsPage($page: Int = 1, $perPage: Int = 15) {
  collections(page: $page, perPage: $perPage) {
    pageNumber
    totalPages
    totalItems
    items {
      id
      name
      tags {
        ...Tag_tag
        id
      }
      countries {
        ...CountryBadge_country
        code
      }
      author {
        displayName
      }
    }
  }
}

query CollectionPage($id: String!) {
  collection(id: $id) {
    id
    name
    createdAt
    updatedAt
    description
    author {
      displayName
    }
    countries {
      code
      ...CountryBadge_country
    }

    tags {
      id
      ...Tag_tag
    }

    elements {
      items {
        ...CollectionElementsTable_element
      }
    }

    ...CollectionActionsMenu_collection
  }
}
