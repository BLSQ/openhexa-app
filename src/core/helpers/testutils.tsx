import { MockedProvider, MockedResponse } from "@apollo/client/testing";
import { faker } from "@faker-js/faker";
import { ByRoleOptions, screen } from "@testing-library/react";
import { ReactNode } from "react";

type TestAppProps = {
  children: ReactNode;
  mocks?: MockedResponse[];
};

jest.mock("next/router", () => require("next-router-mock"));
// This is needed for mocking 'next/link':
jest.mock("next/dist/client/router", () => require("next-router-mock"));
jest.mock("identity/hooks/useMe");

export function TestApp(props: TestAppProps) {
  return (
    <MockedProvider addTypename={false} mocks={props.mocks ?? []}>
      {props.children}
    </MockedProvider>
  );
}

export const LOGGED_IN_USER = {
  id: faker.datatype.uuid(),
  firstName: faker.name.firstName,
  lastName: faker.name.lastName,
  avatar: {
    initials: "AE",
    color: "red",
  },
};

export async function waitForDialog(options?: ByRoleOptions) {
  const dialog = await screen.getByRole("dialog", options);
  return dialog;
}
