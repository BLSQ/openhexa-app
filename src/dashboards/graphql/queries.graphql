query DashboardPage {
  totalNotebooks
  catalog {
    totalItems
  }
  dags {
    totalItems
  }
  lastActivities {
    description
    occurredAt
    url
    status
  }
}
