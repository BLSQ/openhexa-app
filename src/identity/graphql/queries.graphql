query GetUser {
  me {
    hasTwoFactorEnabled
    permissions {
      adminPanel
      superUser
      createWorkspace
    }
    features {
      code
      config
    }
    user {
      ...UserAvatar_user
      email
      id
      firstName
      lastName
      displayName
      avatar {
        initials
        color
      }
    }
  }
}

query AccountPage {
  me {
    hasTwoFactorEnabled
    user {
      firstName
      lastName
      dateJoined
      ...User_user
    }
  }
}
