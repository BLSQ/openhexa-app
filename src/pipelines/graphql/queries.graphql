query PipelinesPage {
  dags {
    totalPages
    totalItems
    items {
      id
      code
      runs(orderBy: EXECUTION_DATE_DESC, perPage: 1) {
        items {
          id
          status
          executionDate
          ...PipelineRunStatusBadge_dagRun
        }
      }
    }
  }
}

query PipelinePage($id: String!, $page: Int) {
  dag(id: $id) {
    id
    code
    schedule
    externalUrl
    template {
      code
      description
      sampleConfig
    }
    schedule
    user {
      ...UserProperty_user
    }
    runs(page: $page, perPage: 15) {
      totalItems
      totalPages
      items {
        id
        triggerMode
        externalId
        externalUrl
        user {
          ...UserColumn_user
        }
        status
        executionDate
        lastRefreshedAt
        duration
        ...PipelineRunStatusBadge_dagRun
      }
    }
  }
}

query PipelineRunPage($pipelineId: String!, $runId: String!) {
  dagRun(id: $runId) {
    id
    externalId
    externalUrl
    executionDate

    status
    config
    ...RunMessages_dagRun
    ...RunLogs_dagRun
    ...PipelineRunStatusBadge_dagRun
  }

  dag(id: $pipelineId) {
    id
    code
  }
}

query PipelineConfigureRunPage($pipelineId: String!) {
  dag(id: $pipelineId) {
    id
    code
    template {
      sampleConfig
      description
    }
    ...PipelineRunForm_dag
  }
}
