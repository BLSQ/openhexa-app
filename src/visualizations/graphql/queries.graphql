query VisualizationsPage($page: Int, $perPage: Int = 15) {
  externalDashboards(page: $page, perPage: $perPage) {
    totalPages
    totalItems
    items {
      id
      name
      url
      description
      ...VisualizationPicture_visualization
      createdAt
      updatedAt
      countries {
        ...CountryBadge_country
      }
      tags {
        ...Tag_tag
      }
    }
  }
}

query Visualization($id: UUID!) {
  externalDashboard(id: $id) {
    id
    name
    url
    description
    ...VisualizationPicture_visualization
    createdAt
    updatedAt
    countries {
      ...CountryBadge_country
    }
    tags {
      ...Tag_tag
    }
  }
}
