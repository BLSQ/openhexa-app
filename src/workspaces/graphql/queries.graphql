query WorkspacesPage {
  workspaces(page: 1, perPage: 1) {
    items {
      slug
    }
  }
}

query WorkspacePage($slug: String!) {
  workspace(slug: $slug) {
    slug
    name
    description
    countries {
      code
      flag
      name
    }
    permissions {
      delete
      update
      manageMembers
    }
    ...ArchiveWorkspace_workspace
    ...InviteMemberWorkspace_workspace
    ...UpdateWorkspaceDescription_workspace
    ...WorkspaceLayout_workspace
  }
}

query WorkspacePipelinesPage(
  $workspaceSlug: String!
  $page: Int
  $perPage: Int
) {
  workspace(slug: $workspaceSlug) {
    slug
    name
    ...WorkspaceLayout_workspace
    ...CreatePipelineDialog_workspace
  }

  pipelines(workspaceSlug: $workspaceSlug, page: $page, perPage: $perPage) {
    items {
      ...PipelineCard_pipeline
    }
    totalItems
    totalPages
    pageNumber
  }
}
query WorkspaceNotebooksPage($workspaceSlug: String!) {
  notebooksUrl
  workspace(slug: $workspaceSlug) {
    slug
    permissions {
      launchNotebookServer
    }
    ...WorkspaceLayout_workspace
  }
}

query WorkspacePipelinePage(
  $workspaceSlug: String!
  $pipelineCode: String!
  $page: Int = 1
  $perPage: Int = 10
) {
  workspace(slug: $workspaceSlug) {
    slug
    name
    ...WorkspaceLayout_workspace
  }

  pipeline: pipelineByCode(workspaceSlug: $workspaceSlug, code: $pipelineCode) {
    ...RunPipelineDialog_pipeline
    ...PipelineVersionPicker_pipeline
    permissions {
      run
      update
      schedule
      delete
    }
    webhookUrl
    webhookEnabled
    id
    code
    name
    description
    schedule
    currentVersion {
      id
      name
      description
      externalLink
      ...PipelineVersionPicker_version
      ...PipelineVersionParametersTable_version
      ...DownloadPipelineVersion_version
    }
    recipients {
      user {
        id
        displayName
      }
    }
    runs(page: $page, perPage: $perPage) {
      items {
        id
        version {
          name
          createdAt
          user {
            ...User_user
          }
        }
        executionDate
        duration
        triggerMode

        user {
          ...UserColumn_user
        }
        ...PipelineRunStatusBadge_run
      }
      totalItems
      totalPages
      pageNumber
    }
  }
}

query WorkspacePipelineVersionsPage(
  $workspaceSlug: String!
  $pipelineCode: String!
  $page: Int!
  $perPage: Int!
) {
  workspace(slug: $workspaceSlug) {
    slug
    name
    ...WorkspaceLayout_workspace
  }
  pipeline: pipelineByCode(workspaceSlug: $workspaceSlug, code: $pipelineCode) {
    id
    code
    name
    currentVersion {
      id
    }
    versions(page: $page, perPage: $perPage) {
      items {
        ...PipelineVersionCard_version
        id
      }
      totalItems
      totalPages
    }
  }
}

query WorkspacePipelineStartPage($workspaceSlug: String!) {
  workspace(slug: $workspaceSlug) {
    slug
    name
    ...WorkspaceLayout_workspace
  }
}

query WorkspacePipelineRunPage($workspaceSlug: String!, $runId: UUID!) {
  workspace(slug: $workspaceSlug) {
    slug
    name
    ...WorkspaceLayout_workspace
    ...RunOutputsTable_workspace
  }

  pipelineRun(id: $runId) {
    id
    version {
      name
      parameters {
        code
        name
        help
        type
        default
        required
        choices
        multiple
      }
    }
    timeout
    config
    executionDate
    duration
    triggerMode
    sendMailNotifications
    pipeline {
      id
      code
      name
      permissions {
        stopPipeline
      }
      ...RunPipelineDialog_pipeline
    }
    user {
      ...User_user
    }
    stoppedBy {
      ...User_user
    }
    ...RunOutputsTable_run
    ...RunPipelineDialog_run
    ...RunMessages_run
    ...RunLogs_run
    ...PipelineRunStatusBadge_run
  }
}

query WorkspaceDatasetsPage(
  $workspaceSlug: String!
  $page: Int
  $perPage: Int
  $query: String
) {
  workspace(slug: $workspaceSlug) {
    slug
    name
    ...WorkspaceLayout_workspace
    ...CreateDatasetDialog_workspace
    permissions {
      createDataset
    }
    pinnedDatasets: datasets(pinned: true, page: 1, perPage: 6) {
      items {
        ...DatasetCard_link
      }
    }
    datasets(query: $query, page: $page, perPage: $perPage) {
      items {
        ...PinDatasetButton_link
        id
        dataset {
          id
          name
          slug
          description
          updatedAt
          workspace {
            slug
            name
          }
          permissions {
            update
            delete
          }
          createdBy {
            ...User_user
          }
        }
      }
      totalItems
      totalPages
      pageNumber
    }
  }
}

query WorkspaceDatasetPage(
  $workspaceSlug: String!
  $datasetSlug: String!
  $versionId: ID!
  $isSpecificVersion: Boolean!
) {
  datasetLink: datasetLinkBySlug(
    workspaceSlug: $workspaceSlug
    datasetSlug: $datasetSlug
  ) {
    id
    ...PinDatasetButton_link
    ...UploadDatasetVersionDialog_datasetLink
    workspace {
      slug
      name
      ...WorkspaceLayout_workspace
    }
    dataset {
      id
      name
      slug
      description
      updatedAt
      workspace {
        slug
        name
      }
      ...DeleteDatasetTrigger_dataset
      ...DatasetLinksDataGrid_dataset
      ...DatasetVersionPicker_dataset
      createdBy {
        ...User_user
      }
      createdAt
      latestVersion {
        createdBy {
          displayName
        }
        createdAt
        ...DatasetVersionFilesDataGrid_version
        ...DatasetVersionPicker_version
      }
      version(id: $versionId) @include(if: $isSpecificVersion) {
        createdBy {
          displayName
        }
        createdAt
        ...DatasetVersionFilesDataGrid_version
        ...DatasetVersionPicker_version
      }

      permissions {
        update
        delete
        createVersion
      }
    }
  }
}

query WorkspaceFilesPage(
  $workspaceSlug: String!
  $page: Int!
  $perPage: Int!
  $prefix: String!
  $query: String
  $ignoreHiddenFiles: Boolean
) {
  workspace(slug: $workspaceSlug) {
    slug
    name
    ...BucketExplorer_workspace
    ...WorkspaceLayout_workspace
    ...UploadObjectDialog_workspace
    ...CreateBucketFolderDialog_workspace
    ...BucketExplorer_workspace

    bucket {
      objects(
        page: $page
        prefix: $prefix
        perPage: $perPage
        query: $query
        ignoreHiddenFiles: $ignoreHiddenFiles
      ) {
        ...BucketExplorer_objects
      }
    }
    permissions {
      createObject
    }
  }
}

query WorkspaceDatabasesPage(
  $workspaceSlug: String!
  $page: Int
  $perPage: Int
) {
  workspace(slug: $workspaceSlug) {
    slug
    name
    permissions {
      update
    }
    database {
      tables(page: $page, perPage: $perPage) {
        totalPages
        totalItems
        items {
          name
          count
        }
      }
    }
    ...DatabaseVariablesSection_workspace
    ...WorkspaceLayout_workspace
  }
}

query WorkspaceDatabaseTablePage($workspaceSlug: String!, $tableName: String!) {
  workspace(slug: $workspaceSlug) {
    slug
    name
    database {
      table(name: $tableName) {
        name
        count
        columns {
          name
          type
        }
        ...DatabaseTableDataGrid_table
      }
    }
    ...DatabaseTableDataGrid_workspace
    ...WorkspaceLayout_workspace
  }
}

query ConnectionsPage($workspaceSlug: String!) {
  workspace(slug: $workspaceSlug) {
    slug
    name
    permissions {
      update
      createConnection
    }
    ...CreateConnectionDialog_workspace
    connections {
      id
      description
      name
      type
      slug
      updatedAt
      permissions {
        update
        delete
      }
    }
    ...WorkspaceLayout_workspace
  }
}

query ConnectionPage($workspaceSlug: String!, $connectionId: UUID!) {
  workspace(slug: $workspaceSlug) {
    slug
    name
    permissions {
      update
    }
    ...WorkspaceLayout_workspace
  }
  connection(id: $connectionId) {
    id
    name
    slug
    description
    type
    createdAt
    permissions {
      update
      delete
    }
    ...ConnectionUsageSnippets_connection
    ...ConnectionFieldsSection_connection
  }
}

query CheckWorkspaceAvailability($slug: String!) {
  workspace(slug: $slug) {
    slug
  }
}
