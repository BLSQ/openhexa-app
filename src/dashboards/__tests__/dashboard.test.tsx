import { render, screen } from "@testing-library/react";
import { TestApp } from "core/helpers/testutils";
import { DashboardPageDocument } from "dashboards/graphql/queries.generated";
import { DateTime, Settings } from "luxon";
import DashboardPage from "pages";

describe("Dashboard", () => {
  it("renders the dashboards' page", async () => {
    (Settings.now as jest.Mock).mockReturnValue(
      DateTime.fromObject({ year: 2022, month: 10, day: 22 }).toMillis(),
    );
    const graphqlMocks = [
      {
        request: {
          query: DashboardPageDocument,
        },
        result: {
          data: {
            totalNotebooks: 0,
            catalog: {
              totalItems: 6,
            },
            dags: {
              totalItems: 0,
            },
            lastActivities: [
              {
                description: "All datasources are up to date!",
                occurredAt: "2022-11-09T00:00:39.651Z",
                url: "/catalog/",
                status: "SUCCESS",
              },
            ],
          },
        },
      },
    ];

    const { container } = render(
      <TestApp mocks={graphqlMocks}>
        <DashboardPage />
      </TestApp>,
    );
    const elm = await screen.findByText("Overview");
    expect(elm).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
