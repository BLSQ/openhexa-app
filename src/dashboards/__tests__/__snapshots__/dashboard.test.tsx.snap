// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders the dashboards' page 1`] = `
<div>
  <main
    class="w-full max-w-7xl mx-auto px-2 sm:px-6 lg:px-8 pb-6"
  >
    <nav
      aria-label="Breadcrumbs"
      class="my-8 px-2"
    >
      <ol
        class="flex items-center"
        role="list"
      >
        <li>
          <div>
            <a
              class="text-gray-400 hover:text-gray-500"
              href="/"
            >
              <svg
                aria-hidden="true"
                class="h-5 w-5 flex-shrink-0"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M9.293 2.293a1 1 0 011.414 0l7 7A1 1 0 0117 11h-1v6a1 1 0 01-1 1h-2a1 1 0 01-1-1v-3a1 1 0 00-1-1H9a1 1 0 00-1 1v3a1 1 0 01-1 1H5a1 1 0 01-1-1v-6H3a1 1 0 01-.707-1.707l7-7z"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="sr-only"
              >
                Home
              </span>
            </a>
          </div>
        </li>
        <li
          class="ml-2"
        >
          <div
            class="flex items-center"
          >
            <svg
              aria-hidden="true"
              class="h-5 w-5 flex-shrink-0 text-gray-300"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.555 17.776l8-16 .894.448-8 16-.894-.448z"
              />
            </svg>
            <span
              class="ml-3 text-sm hover:text-gray-700 font-medium text-gray-500"
            >
              <a
                class=""
                href="/dashboard"
              >
                Dashboard
              </a>
            </span>
          </div>
        </li>
      </ol>
    </nav>
    <div>
      <h2
        class="text-gray-700 font-semibold text-xl mb-4"
      >
        Overview
      </h2>
      <div
        class="grid grid-cols-2 gap-5 sm:grid-cols-3"
      >
        <article
          class="sm:rounded-lg overflow-hidden shadow border-b border-gray-200 bg-white"
        >
          <div
            class="flex gap-4 px-4 py-5"
          >
            <div
              class="flex h-14 w-14 flex-shrink-0 items-center justify-center rounded-lg bg-gray-100 p-2 text-gray-700"
            >
              <svg
                aria-hidden="true"
                class="h-8 w-8"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20.25 6.375c0 2.278-3.694 4.125-8.25 4.125S3.75 8.653 3.75 6.375m16.5 0c0-2.278-3.694-4.125-8.25-4.125S3.75 4.097 3.75 6.375m16.5 0v11.25c0 2.278-3.694 4.125-8.25 4.125s-8.25-1.847-8.25-4.125V6.375m16.5 0v3.75m-16.5-3.75v3.75m16.5 0v3.75C20.25 16.153 16.556 18 12 18s-8.25-1.847-8.25-4.125v-3.75m16.5 0c0 2.278-3.694 4.125-8.25 4.125s-8.25-1.847-8.25-4.125"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </div>
            <dl>
              <dt
                class="truncate text-sm font-medium text-gray-500"
              >
                Datasources
              </dt>
              <dd
                class="text-3xl font-semibold text-gray-700"
              >
                <a
                  class="hover:text-gray-600"
                  href="/catalog"
                >
                  6
                </a>
              </dd>
            </dl>
          </div>
          <div
            class="border-t border-gray-100 bg-gray-50 py-2 px-4 pb-3"
          >
            <a
              class="text-sm text-blue-600 hover:text-blue-500"
              href="/catalog"
            >
              View all
            </a>
          </div>
        </article>
        <article
          class="sm:rounded-lg overflow-hidden shadow border-b border-gray-200 bg-white"
        >
          <div
            class="flex gap-4 px-4 py-5"
          >
            <div
              class="flex h-14 w-14 flex-shrink-0 items-center justify-center rounded-lg bg-gray-100 p-2 text-gray-700"
            >
              <svg
                aria-hidden="true"
                class="h-8 w-8"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 6.042A8.967 8.967 0 006 3.75c-1.052 0-2.062.18-3 .512v14.25A8.987 8.987 0 016 18c2.305 0 4.408.867 6 2.292m0-14.25a8.966 8.966 0 016-2.292c1.052 0 2.062.18 3 .512v14.25A8.987 8.987 0 0018 18a8.967 8.967 0 00-6 2.292m0-14.25v14.25"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </div>
            <dl>
              <dt
                class="truncate text-sm font-medium text-gray-500"
              >
                Notebooks
              </dt>
              <dd
                class="text-3xl font-semibold text-gray-700"
              >
                <a
                  class="hover:text-gray-600"
                  href="/notebooks"
                >
                  0
                </a>
              </dd>
            </dl>
          </div>
          <div
            class="border-t border-gray-100 bg-gray-50 py-2 px-4 pb-3"
          >
            <a
              class="text-sm text-blue-600 hover:text-blue-500"
              href="/notebooks"
            >
              View all
            </a>
          </div>
        </article>
        <article
          class="sm:rounded-lg overflow-hidden shadow border-b border-gray-200 bg-white"
        >
          <div
            class="flex gap-4 px-4 py-5"
          >
            <div
              class="flex h-14 w-14 flex-shrink-0 items-center justify-center rounded-lg bg-gray-100 p-2 text-gray-700"
            >
              <svg
                aria-hidden="true"
                class="h-8 w-8"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.75 3.104v5.714a2.25 2.25 0 01-.659 1.591L5 14.5M9.75 3.104c-.251.023-.501.05-.75.082m.75-.082a24.301 24.301 0 014.5 0m0 0v5.714c0 .597.237 1.17.659 1.591L19.8 15.3M14.25 3.104c.251.023.501.05.75.082M19.8 15.3l-1.57.393A9.065 9.065 0 0112 15a9.065 9.065 0 00-6.23-.693L5 14.5m14.8.8l1.402 1.402c1.232 1.232.65 3.318-1.067 3.611A48.309 48.309 0 0112 21c-2.773 0-5.491-.235-8.135-.687-1.718-.293-2.3-2.379-1.067-3.61L5 14.5"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </div>
            <dl>
              <dt
                class="truncate text-sm font-medium text-gray-500"
              >
                Pipelines
              </dt>
              <dd
                class="text-3xl font-semibold text-gray-700"
              >
                <a
                  class="hover:text-gray-600"
                  href="/pipelines"
                >
                  0
                </a>
              </dd>
            </dl>
          </div>
          <div
            class="border-t border-gray-100 bg-gray-50 py-2 px-4 pb-3"
          >
            <a
              class="text-sm text-blue-600 hover:text-blue-500"
              href="/pipelines"
            >
              View all
            </a>
          </div>
        </article>
      </div>
    </div>
    <div
      class="mt-12"
    >
      <h2
        class="text-gray-700 font-semibold text-xl mb-4"
      >
        Last Activity
      </h2>
      <article
        class="sm:rounded-lg overflow-hidden shadow border-b border-gray-200 bg-white"
      >
        <div>
          <div
            class="overflow-x-auto"
          >
            <div
              class="overflow-x-auto overflow-y-hidden"
            >
              <table
                class="w-full divide-y divide-gray-200 "
                role="table"
              >
                <thead
                  class="bg-gray-50"
                >
                  <tr
                    class=""
                    role="row"
                  >
                    <th
                      class="px-4 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider whitespace-nowrap"
                      colspan="1"
                      role="columnheader"
                      scope="col"
                    >
                      Name
                    </th>
                    <th
                      class="px-4 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider whitespace-nowrap"
                      colspan="1"
                      role="columnheader"
                      scope="col"
                    >
                      Status
                    </th>
                    <th
                      class="px-4 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider whitespace-nowrap"
                      colspan="1"
                      role="columnheader"
                      scope="col"
                    >
                      Date
                    </th>
                    <th
                      class="px-4 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider whitespace-nowrap"
                      colspan="1"
                      role="columnheader"
                      scope="col"
                    />
                  </tr>
                </thead>
                <tbody
                  class="divide-y divide-gray-200"
                  role="rowgroup"
                >
                  <tr
                    class=""
                    role="row"
                  >
                    <td
                      class="px-4 py-5 text-sm text-gray-500 h-full max-w-[250px] text-sm text-gray-900 lg:max-w-[400px] whitespace-nowrap"
                      role="cell"
                    >
                      <div
                        class="truncate"
                      >
                        All datasources are up to date!
                      </div>
                    </td>
                    <td
                      class="px-4 py-5 text-sm text-gray-500 h-full whitespace-nowrap"
                      role="cell"
                    >
                      <span
                        class="inline-flex items-center truncate whitespace-nowrap rounded-md border border-gray-500 border-opacity-10 font-medium px-2 py-0.5 text-xs bg-emerald-50 text-emerald-500 flex items-center"
                      >
                        Succeeded
                      </span>
                    </td>
                    <td
                      class="px-4 py-5 text-sm text-gray-500 h-full whitespace-nowrap"
                      role="cell"
                    >
                      <time
                        class="whitespace-nowrap truncate"
                        datetime="2022-11-09T01:00:39.651+01:00"
                        title="2022-11-09T01:00:39.651+01:00"
                      >
                        in 18 days
                      </time>
                    </td>
                    <td
                      class="px-4 py-5 text-sm text-gray-500 h-full whitespace-nowrap"
                      role="cell"
                    >
                      <div
                        class="w-full"
                      >
                        <a
                          class="flex w-full cursor-pointer items-center justify-end outline-none text-blue-600 hover:text-blue-500"
                          href="/catalog"
                        >
                          View
                          <svg
                            aria-hidden="true"
                            class="inline h-5"
                            fill="currentColor"
                            viewBox="0 0 20 20"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              clip-rule="evenodd"
                              d="M7.21 14.77a.75.75 0 01.02-1.06L11.168 10 7.23 6.29a.75.75 0 111.04-1.08l4.5 4.25a.75.75 0 010 1.08l-4.5 4.25a.75.75 0 01-1.06-.02z"
                              fill-rule="evenodd"
                            />
                          </svg>
                        </a>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </article>
    </div>
  </main>
</div>
`;
